src/parser/ast_node/mod.rs

```diff
@@ -0,0 +1,8 @@
+pub mod ast_node;
+mod ast_node_concat;
+mod ast_node_group;
+mod ast_node_literal;
+mod ast_node_optional;
+mod ast_node_plus;
+mod ast_node_star;
+mod ast_node_union;
```

Comment:
We might need them to be public if we implement nfa in a separate mod

Reply:
we can do this when we need to do this